packages/vite/src/node/build.ts

```diff
@@ -372,12 +372,14 @@ async function doBuild(
   )
 
   const resolve = (p: string) => path.resolve(config.root, p)
-  const input = libOptions
+  let input = libOptions
     ? resolve(libOptions.entry)
     : typeof options.ssr === 'string'
     ? resolve(options.ssr)
     : options.rollupOptions?.input || resolve('index.html')
-
+  if (input === 'index.html') {
+    input = resolve(input)
+  }
```

Comment:
We shouldn't have a particular case for `index.html`. This will be still broken for `foo.html`. We need to check where input is being used which results in the paths being different on both platforms. Probably in the `html` plugin, when computing these paths. I think we may be missing a `normalizePath`. https://github.com/vitejs/vite/blob/main/packages/vite/src/node/plugins/html.ts#L929